app/models/bike.rb

```diff
@@ -652,67 +653,46 @@ def registration_location
+  # 2. The current abandoned record, if one is present
+  # 3. The current stolen record, if one is present
+  # 4. The creation organization, if one is present
+  # 5. The bike owner's address, if available
+  # 6. The request's IP address, if given
+  def find_location_info(geolocation = nil)
+    location_record = [
+      current_parking_notification,
+      current_stolen_record,
+      creation_organization,
+      owner,
+    ].compact.find(&:bike_location_info?)
+
+    return location_record if location_record.present?
+
+    # Need to test what the responses from this look like - probably slightly
+    # different
+    geolocation&.address = current_parking_notification&.address
+    geolocation if geolocation&.bike_location_info?
+  end
+
+  # Set the bike's location data (lat/long, city, postal code, country, etc.).
   def set_location_info(request_location: nil)
+    # ensure current_stolen_record is set if available
     find_current_stolen_record
-    if location_info_present?(current_parking_notification)
-      self.latitude = current_parking_notification.latitude
-      self.longitude = current_parking_notification.longitude
-      self.city = current_parking_notification.city
-      self.country = current_parking_notification.country
-      self.zipcode = current_parking_notification.zipcode
-      self.address = current_parking_notification.address
-    elsif location_info_present?(current_stolen_record)
-      self.latitude = current_stolen_record.latitude
-      self.longitude = current_stolen_record.longitude
-      self.city = current_stolen_record.city
-      self.country = current_stolen_record.country
-      self.zipcode = current_stolen_record.zipcode
-      self.address = current_stolen_record.display_address(force_show_address: true)
-    elsif location_info_present?(creation_organization)
-      self.latitude = creation_organization.location_latitude
-      self.longitude = creation_organization.location_longitude
-      self.city = creation_organization.city
-      self.country = creation_organization.country
-      self.zipcode = creation_organization.zipcode
-      # This is a hack, need to fix
-      self.address = [city, zipcode, country&.name].select(&:present?).join(" ").presence
-    elsif location_info_present?(owner)
-      self.latitude = owner.latitude
-      self.longitude = owner.longitude
-      self.city = owner.city
-      self.country = owner.country
-      self.zipcode = owner.zipcode
-      self.address = owner.address
-    elsif location_info_present?(request_location)
-      self.latitude = request_location.latitude
-      self.longitude = request_location.longitude
-      self.city = request_location.city
-      self.country = Country.fuzzy_find(request_location&.country_code)
-      self.zipcode = request_location.postal_code
-      # Need to test what the responses from this look like - probably slightly different
-      # self.address = current_parking_notification.address
```

Comment:
L714 (uncommented) is here in case we need it but I can also kill it: https://github.com/bikeindex/bike_index/pull/1546/files#diff-998a6bdfbdc9035d6f2c1a089867c8efR677

Reply:
Having worked with this now for a little while, I'm not sure there is a strong benefit to having an address column. Can't we just generate it when it needs to be displayed?

I don't think it has a drawback to being there, other than the fact that we have `street`, `address` and `display_address` methods/attributes - which is confusing.

Reply:
> Can't we just generate it when it needs to be displayed?

That's the previous behavior, but for the purposes of simplifying the `Geocodeable` implementation (in particular the `geocode_columns` method below) we pre-generate and store an address. 
```rb
# app/models/concerns/geocodeable.rb L16-36 (2b78f947)

  # Customize by overwriting in the Geocodeable model.
  def geocode_data
    @geocode_data ||= address
  end

  # An array of symbols of the db columns upon which `geocode_data`
  # depends.These are checked for changes before geocoding using
  # `should_be_geocoded?`. By default, empty, so we don't check for changes.
  def geocode_columns; []; end

  def any_geocode_columns_changed?
    return true if geocode_columns.blank?
    geocode_columns.any? { |col| public_send("#{col}_changed?") }
  end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/2b78f947/app/models/concerns/geocodeable.rb#L16-L36)]</sup>

But if the set of fields for `Geocodeable` are common by convention, we can alternatively drop `address` and specify `geocode_columns` in `Geocodeable`.

Reply:
> But if the set of fields for `Geocodeable` are common by convention, we can alternatively drop `address` and specify `geocode_columns` in `Geocodeable`.

I _think_ this would be good. Do you think it would make things simpler or more complicated? If it's ultimately going to make things more confusing, it's not worth doing. The only reason to do it would be to make things easier to understand, I don't think there is anything actually wrong with the current implementation.

Might make sense to work on this in a different PR, this PR seems 🎉, but  ¯\\\_(ツ)\_/¯

Reply:
That makes sense — that adds more logic to `Geocodeable` but reduces duplication overall. 27e79db6 drops the previously added `address` columns and extracts address-string-building logic to a single `Geocodeable` class method. 

<details>
<summary><code>Geocodeable.address</code></summary>

```rb
# app/models/concerns/geocodeable.rb L41-84 (d3d9d0af)

# Build an address string from the given object's location data.
#
# The following keyword args accept booleans for inclusion / omission in the
# string: `street` `city` `state` `zipcode`, `country`.
#
# By default, the country is included as an ISO abbreviation and is required
# (an empty string is returned if no country is available.)
#
# The `country` keyword arg also accepts a list of options to customize
# output:
#
# - :iso or :name for the format
# - :optional to make the country optional
# - :skip_default to omit the country name if it's the default country (US)
#
# Returns a String.
def self.address(obj, street: true, city: true, state: true, zipcode: true, country: [:iso])
  return "" if obj.blank?

  include_country =
    country && !(obj.country&.default? && country.include?(:skip_default))

  country_name =
    if include_country
      country_format = country.find { |e| e.in? %i[iso name] } || :iso
      country_name = obj.country&.public_send(country_format)

      country_is_required = !country.include?(:optional)
      not_enough_info = country_is_required && country_name.blank?
      return "" if not_enough_info

      country_name
    end

  [
    street && obj.street,
    city && obj.city,
    [
      state && obj.state&.abbreviation,
      zipcode && obj.zipcode,
    ].reject(&:blank?).join(" "),
    country_name,
  ].reject(&:blank?).join(", ")
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/d3d9d0af/app/models/concerns/geocodeable.rb#L41-L84)]</sup>
</details>